File: src/BHS.Contracts/Blog/PostRequest.cs
using System.ComponentModel.DataAnnotations;

namespace BHS.Contracts.Blog;

public record PostRequest(
    string Title,
    string ContentMarkdown,
    DateTimeOffset DatePublished,
    IEnumerable<Category> Categories,
    [Url] Uri? FilePath = null,
    string? PhotosAlbumSlug = null,
    Author? Author = null);

File: src/bhs-web-angular-app/src/app/data/blog/models/post-request.ts
import { Category } from './category';
import { Author } from '@data/authors';

export interface PostRequest {
  title: string;
  contentMarkdown: string;
  filePath: string | null;
  photosAlbumSlug: string | null;
  author: Author | null;
  datePublished: Date;
  categories: Array<Category>;
}
